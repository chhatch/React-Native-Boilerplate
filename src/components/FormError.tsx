import React from 'react'
import { StyleProp, Text, View } from 'react-native'
import PropTypes, { InferProps } from 'prop-types'

//this will be done with a proper type, but this is the syntax for this approach
const FormError = ({isError, message, style}: {isError: bool, message: string, style: StyleProp}) => {
    return <Text style={style}>{isError ? message : ' '}</Text>
}

export default FormError
