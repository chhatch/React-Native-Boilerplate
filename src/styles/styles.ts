import { Dimensions, StyleSheet, TextStyle, ViewStyle } from 'react-native'

const width = Dimensions.get('window').width
const height = Dimensions.get('window').height

export const styles = StyleSheet.create({
    input: {
        height: 50,
        backgroundColor: '#ededed',
        width: '33%',
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    button: {
        marginBottom: 20,
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
    },
    separator: {
        marginVertical: 30,
        height: 1,
        width: '80%',
    },
    head: { height: 40, backgroundColor: '#f1f8ff' },
    text: { margin: 6 },
    loading: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'center',
    },
    formError: {
        marginBottom: 20,
    },
})

interface Alerts {
    alertContainer: ViewStyle
    error: TextStyle
    success: TextStyle
}

export const alertStyles = StyleSheet.create<Alerts>({
    alertContainer: {
        marginTop: 63,
        position: 'absolute',
        width: width,
        zIndex: 999,
    },
    error: {
        backgroundColor: 'red',
        fontSize: 20,
    },
    success: {
        backgroundColor: 'green',
        fontSize: 20,
    },
})
