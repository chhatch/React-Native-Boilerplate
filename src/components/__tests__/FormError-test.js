import * as React from 'react'
import renderer from 'react-test-renderer'

import FormError from '../FormError'
import { styles } from '../../styles/styles'

const message = 'Snapshot test!'
const style = styles.formError

it(`renders an error`, () => {
    const tree = renderer.create(<FormError isError={true} message={message} style={style} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it(`does not render an error`, () => {
    const tree = renderer.create(<FormError isError={false} message={message} style={style} />).toJSON()
    expect(tree).toMatchSnapshot()
})
