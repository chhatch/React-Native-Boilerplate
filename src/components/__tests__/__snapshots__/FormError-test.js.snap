// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error 1`] = `
<Text
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
   
</Text>
`;

exports[`renders an error 1`] = `
<Text
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  Snapshot test!
</Text>
`;
